etrago/cluster/networkclustering.py

```diff
@@ -517,43 +552,9 @@ def normed(x):
             line_length_factor=line_length_factor)
         network = clustering.network
 
-    # define weighting based on conventional 'old' generator spatial
-    # distribution
-    non_conv_types = {
-        'biomass',
-        'wind_onshore',
-        'wind_offshore',
-        'solar',
-        'geothermal',
-        'load shedding',
-        'extendable_storage'}
-    # Attention: network.generators.carrier.unique()
-    gen = (network.generators.loc[(network.generators.carrier
-                                   .isin(non_conv_types) == False)]
-           .groupby('bus').p_nom.sum()
-                                .reindex(network.buses.index, fill_value=0.) +
-           network.storage_units
-                                .loc[(network.storage_units.carrier
-                                      .isin(non_conv_types) == False)]
-                  .groupby('bus').p_nom.sum()
-                  .reindex(network.buses.index, fill_value=0.))
-
-    load = network.loads_t.p_set.mean().groupby(network.loads.bus).sum()
+        weight = weight.groupby(busmap.values).sum()
```

Comment:
why is the line now additionally necessary?

Reply:
When buses are deleted in remove_stubs clustering, there is still a weighting for the deleted stubs otherwise. In this line, the weightings of the stubs are, like the generators,  aggregated to the next remaining bus